### PR TITLE
Implement correct behavior on unhandled exception

### PR DESCRIPTION
**Pull request by ludovic-henry**

### CODE REVIEW

File: mono/metadata/domain.c [hunk 1]
```diff
@@ -725,6 +725,10 @@ mono_init_internal (const char *filename, const char *exe_filename, const char *
                 mono_defaults.corlib, "System.Threading", "ThreadAbortException");
 	g_assert (mono_defaults.threadabortexception_class != 0);
 
+	mono_defaults.appdomainunloadedexception_klass = mono_class_from_name (
+	            mono_defaults.corlib, "System", "AppDomainUnloadedException");
+	g_assert (mono_defaults.appdomainunloadedexception_klass);
```
Comment (tritao) [inline mono/metadata/domain.c]: Please make sure this is kept by the linker in Monodroid and Monotouch.

Comment (marek-safar) [inline mono/metadata/domain.c]: It should not be needed as they don't support appdomains

**Comment by marek-safar** [comment]
Any tests for correct behaviour?

**Comment by kumpera** [comment]
Any explanation what the correct behavior is?

The commit messages have no information on what is been changed.

This can't be reviewed without a proper explanation of what it is trying to change.

File: mono/metadata/domain.c (hunk 1 above)
Comment (tritao) [inline mono/metadata/domain.c]: The code is asserting that the class exists though, so the runtime will blow up nonetheless.

**Comment by ludovic-henry** [comment]
After some testing (https://gist.github.com/ludovic-henry/02fcb926a0dcd3e83578) the 4.2 behavior is already the correct one. It also has been checked on windows against .NET 4.5, and the current behavior match with it.

**Pull request closed without merging**